### PR TITLE
Update behaviortree_cpp to v4

### PR DESCRIPTION
**Pull request by bektaskemal**

### CODE REVIEW

File: dependencies.repos [hunk 1]
```diff
@@ -45,3 +45,7 @@ repositories:
     type: git
     url: https://github.com/BehaviorTree/Groot.git
     version: master
+  behavior_tree_cpp:
```
Comment (sea-bass) [inline dependencies.repos:48]: For consistency

```suggestion
  behaviortree_cpp:
```

Also, can you place this above Groot and replace the comment to say something like:

"BehaviorTree.CPP and Groot for visualization"

File: tb3_autonomy/bt_xml/nav_tree_naive.xml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <!-- Behavior tree that sequentially navigates locations naively -->
-<root main_tree_to_execute = "MainTree" >
+<root BTCPP_format="4"main_tree_to_execute = "MainTree" >
```
Comment (sea-bass) [inline tb3_autonomy/bt_xml/nav_tree_naive.xml:2]: For all these trees, looking at the [tutorial](https://www.behaviortree.dev/docs/tutorial-basics/tutorial_01_first_tree#create-a-tree-dynamically-with-an-xml), it seems like `main_tree_to_execute` isn't needed, so can we remove these?

```suggestion
<root BTCPP_format="4">
```

**Review by sea-bass — COMMENTED** [review]
This is amazing, and worked for me great. 

Thank you for your contribution, and I'm good after some minor comments.

**Comment by bektaskemal** [comment]
> This is amazing, and worked for me great.
> 
> Thank you for your contribution, and I'm good after some minor comments.

Hey thanks, updated as you asked.

**Review by sea-bass — APPROVED** [review]
Thank you!

**Pull request merged**